client/consumables.lua

```diff
@@ -129,329 +116,325 @@ local function CokeBaggyEffect()
             RestorePlayerStamina(PlayerId(), 1.0)
         end
         startStamina -= 1
-        if math.random(1, 100) < 10 and IsPedRunning(ped) then
-            SetPedToRagdoll(ped, math.random(1000, 3000), math.random(1000, 3000), 3, false, false, false)
+        if math.random(1, 100) < 10 and IsPedRunning(cache.ped) then
+            SetPedToRagdoll(cache.ped, math.random(1000, 3000), math.random(1000, 3000), 3, false, false, false)
         end
         if math.random(1, 300) < 10 then
             AlienEffect()
             Wait(math.random(3000, 6000))
         end
     end
-    if IsPedRunning(ped) then
-        SetPedToRagdoll(ped, math.random(1000, 3000), math.random(1000, 3000), 3, false, false, false)
+    if IsPedRunning(cache.ped) then
+        SetPedToRagdoll(cache.ped, math.random(1000, 3000), math.random(1000, 3000), 3, false, false, false)
     end
     SetRunSprintMultiplierForPlayer(PlayerId(), 1.0)
 end
 
+local function SmokeWeed()
+    CreateThread(function()
+        while SmokingWeed do
+            Wait(10000)
+            TriggerServerEvent('hud:server:RelieveStress', math.random(15, 18))
+            RelieveCount += 1
+            if RelieveCount == 6 then
+                exports.dpemotes:CancelAnimation()
+                if SmokingWeed then
+                    SmokingWeed = false
+                    RelieveCount = 0
+                end
+            end
+        end
```

Comment:
```suggestion
        while SmokingWeed do
            Wait(10000)
            TriggerServerEvent('hud:server:RelieveStress', math.random(15, 18))
            for RelieveCount = 0, 6, 1 do
                if RelieveCount == 6 then
                    exports.dpemotes:CancelAnimation()
                    SmokingWeed = false
                    RelieveCount = 0
                end
            end
        end
```
The while loop already checks ```SmokingWeed```  and a for loop could serve you better here.